### PR TITLE
fixes/SPHEREON-1185_FIDS

### PR DESCRIPTION
**Pull request by sanderPostma**

### CODE REVIEW

File: packages/oid4vci-holder/src/agent/OID4VCIHolder.ts [hunk 1]
```diff
@@ -227,8 +227,8 @@ export class OID4VCIHolder implements IAgentPlugin {
     // "JcsEd25519Signature2020"
   ]
   private readonly didMethodPreferences: Array<SupportedDidMethodEnum> = [
+    SupportedDidMethodEnum.DID_JWK, // FIXME prefer JWK until we devise a method to detect when to use EBSI/jcs for did:key and when not
```
Comment (sanderPostma) [inline packages/oid4vci-holder/src/agent/OID4VCIHolder.ts:230]: Not sure whether this is ok for now

File: packages/oid4vci-holder/src/agent/OID4VCIHolderService.ts [hunk 2]
```diff
@@ -261,14 +261,14 @@ export const getIdentifierOpts = async (args: GetIdentifierArgs): Promise<Manage
...
     })
-    if (created) {
-      await agentContext.agent.emit(OID4VCIHolderEvent.IDENTIFIER_CREATED, { result })
-    }
     identifier = await context.agent.identifierManagedGetByDid({
       identifier: result,
       keyType,
       offlineWhenNoDIDRegistered: result.did.startsWith('did:ebsi:'),
     })
+    if (created) {
```
Comment (BtencateSphereon) [inline packages/oid4vci-holder/src/agent/OID4VCIHolderService.ts:269]: we were passing in a value called result. what happened to that?

File: packages/siopv2-oid4vp-op-auth/src/session/OpSession.ts [hunk 3]
```diff
@@ -309,8 +309,12 @@ export class OpSession {
       const totalInputDescriptors = request.presentationDefinitions?.reduce((sum, pd) => {
         return sum + pd.definition.input_descriptors.length
       }, 0)
+      const totalVCs = args.verifiablePresentations?.reduce((sum, vp) => {
+        const uvp = CredentialMapper.toUniformPresentation(vp)
+        return sum + (uvp.verifiableCredential?.length ?? 0)
+      }, 0)
 
-      if (!request.presentationDefinitions || !args.verifiablePresentations || args.verifiablePresentations.length !== totalInputDescriptors) {
+      if (!request.presentationDefinitions || !args.verifiablePresentations || totalVCs !== totalInputDescriptors) {
```
Comment (BtencateSphereon) [inline packages/siopv2-oid4vp-op-auth/src/session/OpSession.ts:317]: InputDescriptor does not equal 1 vc?

**Review by BtencateSphereon — CHANGES_REQUESTED** [review]

File: packages/oid4vci-holder/src/agent/OID4VCIHolderService.ts (hunk 2 above)
Comment (sanderPostma) [reply packages/oid4vci-holder/src/agent/OID4VCIHolderService.ts:269]: The went inidentifierManagedGetByDid
The mobile wallet expects a managed identifier now, not a string

File: packages/siopv2-oid4vp-op-auth/src/session/OpSession.ts (hunk 3 above)
Comment (sanderPostma) [reply packages/siopv2-oid4vp-op-auth/src/session/OpSession.ts:317]: I guess so, but each VP can have one or more VCs. So two VCs can be in one VP and a third can be in another VP.
"args.verifiablePresentations.length !== totalInputDescriptors" is not handling that.

**Review by BtencateSphereon — APPROVED** [review]

**Pull request merged**